blockchains/eth/lib/next_interval_calculator.js

```diff
@@ -49,11 +49,24 @@ function nextIntervalCalculator(worker) {
   };
 }
 
+/**
+ * Function for calculating the next interval to be used in the worker's methods for querying the node.
+ * @param {BaseWorker} worker A worker instance, inherriting the BaseWorker class.
+ * @returns {object} The interval, derived from the progress of the worker
+ */
+function nextIntervalCalculatorV2(worker) {
```

Comment:
Changing the original:

```js
function nextIntervalCalculator(worker) {
  return {
    fromBlock: worker.lastExportedBlock + 1,
    toBlock: Math.min(worker.lastExportedBlock + worker.settings.BLOCK_INTERVAL, worker.lastConfirmedBlock)
  };
}
```
to
```js
function nextIntervalCalculator(lastExported, lastConfirmed, blockInterval) {
...
 }
```
would allow you to reuse the same code. Also this is considered better design, as it more clearly denotes the intent of the function.